### DISCUSSION TITLE
A new interactive method : drag the windows to different part of the other windows

### DISCUSSION THREAD

**Discussion by MisteryMonster** [FancyWM/fancywm#172] [Ideas]
This is inspired by AltSnap https://github.com/RamonUnch/AltSnap, it can change window resize behavior by clicking different part of the window, so I think if it can be using in this way :

![image](https://user-images.githubusercontent.com/40703811/216376985-1c49b689-be2c-419d-80c8-2de49880800e.png)

or like this:

![image](https://user-images.githubusercontent.com/40703811/216389217-73084702-d506-4fba-88d1-20dc22ec83f7.png)

when dragging a window/panel over windows, the different part of the window will perform different behavior (exclude the float window), and gap is necessary because if the gap are too small, you would only create new vertical/horizontal panel by neighbor windows. and using gap to reposition.

**Comment by riotrah** [comment]
I haven't looked at altsnap  (thanks for sharing it), but if you're able to run Shell commands or scripts for each corner that you snap to, you could invoke fan CWM commands from autohot key. Not the cleanest solution but it might work

**Comment by airtonix** [comment]
yeah nah. feels like a complicated solution to something that doesn't need solving.

PopShell is a much more obvious approach to using window regions because the implied outcome is more obvious: 

https://github.com/FancyWM/fancywm/assets/61225/bbc1a3da-4247-45d5-a50f-e541d4707429

Here you can see that I show what gnome shell does without popshell, then i enable it via the status bar up the top and you can see what happens when i start dragging a window over other windows.

**Comment by veselink1** [comment]
I see the thread has some new comments, so wanted to update. I have the beginnings of an implementation somewhere, I can push to a feature branch. I think it is possible to do, but needs time dedicated to it. 

Ultimately, I like the idea, so if anyone wants to work on it, I'm happy to help, I don't think I can allocate the time to implement it myself.
